app/javascript/map.js

```diff
@@ -85,6 +89,40 @@ class Map {
     }
   }
 
+  placeLocationsInBounds() {
+    setTimeout(() => {
+      googleMap.fitBounds(bounds);
+    }, 2000);
+
+    googleMap.addListener('idle', () => {
+      this.clearMarkers();
```

Comment:
You should only clear old markers after the new markers have been placed.  Funcationally, this wipes the map clean before it gets a response...  Additionally, you shouldn't clear a marker if it is currently open:  Right now, when you click a marker, it resizes the map for the info window, which clears the markers.  Try using it with a production set of data.